### ISSUE TITLE
Staging Review - Accessibility Feedback - Converted components: Button, Button Pair and Modal

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#64497]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

-

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

### Button
No feedback!

### Button pair
- **Must:** At high zoom or small viewports, the buttons stack in a counterintuitive way. The first button in the DOM (left on desktop/standard zoom) is displayed below the second button. This leads to some quirkiness for all users (Yes/No pattern on desktop becomes a No/Yes pattern on mobile), but for keyboard users it leads to a confusing tab order. The first button in the DOM gets the first tab stop, and advancing to the next tab stop has the appearance of going backwards on the page -- and then, if you have another tab stop after that, visually you leapfrog down to it.
   
   I feel like there are three-ish options, with my preferences roughly in this order:
   1. Preserve DOM order in the visual order. If it comes first in the DOM, it comes first on the page.
   2. Have some extra focus management script get triggered at that breakpoint so that the tab order matches the visual order. You still get No/Yes quirkiness but the keyboard interaction is smooth.
   3. Programmatically re-order the elements in the DOM to get the desired visual order, and thus also get the desired tab order.
   
### Modal
- **Must:** When a [modal doesn't have a title](https://design.va.gov/components/modal#without-title), the containing `div` has an `aria-label="undefined modal"` which then gets announced to screen reader users. Needs to be a more graceful way of handling this than saying "undefined" when the title prop is undefined.
   - **Should:** This might solve the "must" feedback --- does it ever make sense for a modal to not have a title? I feel like the [USWDS accessibility guidance](https://designsystem.digital.gov/components/modal/#accessibility-modal) kind of implies that modals should always have a title, though it doesn't say it explicitly. My preference would be for us to codify that in the design system, all modals need to have a title. 
- **Weird bug? / Must:** In Storybook, when you have the [Crisis Line modal](https://design.va.gov/storybook/?path=/docs/uswds-va-modal--default#crisis-line-modal) example open and tab through the elements inside the modal, the page behind the modal scrolls down. That unexpected motion could be disorienting, and might be confusing when you close the modal. But it's just on the one example, which makes me think it might be a weird bug with that modal in Storybook rather than a problem with the component.
- **Consider:** There's been some movement toward using visible "CLOSE" text alongside the "X" button in other contexts, like the notification and card components. I'd love to see that happen here too!

**Comment by humancompanion-usds** [comment]
@caw310 - It's unclear to me which of these were going to tackle in this bug. Just the MUST feedback? All of it? I think we should split this up into discrete units to better track the work given that this spans multiple components.

**Comment by caw310** [comment]
@humancompanion-usds , I can split this into 2 tickets - one for each component. SInce we are having staging reviews for multiple components together, the feedback is being merged. Going forward, we can separate out staging reviews or ask Brian to create separate tickets.

**Comment by caw310** [comment]
I created two new tickets to separate the work needed for each component. 
Button pair - https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2086
Modal - https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2087

**Comment by caw310** [comment]
Closing as the musts have been addressed in other tickets.
